libfuse/fs.go

```diff
@@ -220,11 +220,24 @@ type Root struct {
 	public  *FolderList
 }
 
+var _ fs.NodeAccesser = (*FolderList)(nil)
+
+// Access implements fs.NodeAccesser interface for *Root.
+func (*Root) Access(ctx context.Context, r *fuse.AccessRequest) error {
+	if int(r.Uid) != os.Getuid() {
+		// short path: not accessible by anybody other than the logged in user.
+		// This is in case we enable AllowOther in the future.
+		return fuse.EPERM
+	}
+
+	return nil
```

Comment:
You should return an error for anything asking about write permissions in the root.